src/util/taproot.rs

```diff
@@ -1059,6 +1096,8 @@ mod test {
     fn control_block_verify() {
         let secp  = Secp256k1::verification_only();
         // test vectors obtained from printing values in feature_taproot.py from bitcoin core
+        // excluding four test cases using invalid leaf version in the control block
+        // see details in <https://github.com/rust-bitcoin/rust-bitcoin/pull/718#issuecomment-1007596228>
```

Comment:
No need to exclude these tests, if you change the above, the tests can then be uncommented.

Reply:
Regardless they should definitely not be called **invalid** if they are not actually invalid.

Sorry for contributing to confusion instead of just checking the BIP myself.

Reply:
@apoelstra sorry, now I do not understand at all what should be done with these tests or `LeafVersion` type.

Reply:
@dr-orlovsky the tests should be restored. We should only reject odd values or `0x50`, as the BIP says in footnote 7. Nothing else should be rejected.

Reply:
How to classify all values which are not `0xC0`, `LeafVersion::Future`?

Reply:
Everything that isn't known or invalid is `Future` (even though the real-life future will not include BIPs that ignore the advice in BIP 341).

Reply:
done